File: backend/src/main/java/com/ftn/activityapp/dto/RecommendationResponse.java
package com.ftn.activityapp.dto;

import com.ftn.activityapp.enums.DailyState;
import com.ftn.activityapp.enums.IntensityLevel;
import com.ftn.activityapp.enums.RecommendationType;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Builder
public class RecommendationResponse {
    private Long id;
    private Long userId;
    private LocalDate date;
    private DailyState dailyState;
    private RecommendationType recommendationType;
    private IntensityLevel intensity;
    private Integer durationMinutes;
    private String message;

    private String notification;

    private NutritionResponseDto nutrition;
    private FreeTimeResponseDto freeTime;
    private WellnessResponseDto wellness;
    private MotivationResponseDto motivation;
}

File: mobile-app/app/src/main/java/com/example/activityapp/data/remote/dto/RecommendationResponse.kt
package com.example.activityapp.data.remote.dto

import com.example.activityapp.data.remote.dto.freetime.FreeTimeResponseDto
import com.example.activityapp.data.remote.dto.nutrition.NutritionResponse
import com.example.activityapp.data.remote.dto.planner.PlannerSuggestionDto
import com.example.activityapp.data.remote.dto.wellness.WellnessResponse

data class RecommendationResponse(
    val id: Long,
    val userId: Long,
    val date: String,
    val dailyState: String,
    val recommendationType: String,
    val intensity: String,
    val durationMinutes: Int,
    val message: String,
    val notification: String?,
    val nutrition: NutritionResponse?,
    val freeTime: FreeTimeResponseDto?,
    val wellness: WellnessResponse?,
    val motivation: MotivationResponse?,
    val plannerSuggestions: List<PlannerSuggestionDto> = emptyList()

)